### PR TITLE
fix for element cast exception

### PR DESCRIPTION
**Pull request by epankou**

### CODE REVIEW

File: daemon/src/org/eclipse/iofog/process_manager/ContainerManager.java [hunk 1]
```diff
@@ -29,9 +29,8 @@
  */
 public class ContainerManager {
 
+	private static final String EMPTY = "";
```
Comment (epankou) [inline daemon/src/org/eclipse/iofog/process_manager/ContainerManager.java]: string utils empty

File: daemon/src/org/eclipse/iofog/process_manager/ContainerManager.java [hunk 2]
```diff
@@ -116,45 +118,73 @@ private void startElement() {
...
-			LoggingService.logWarning(MODULE_NAME, String.format("error removing container \"%s\"", containerId));
-			throw e;
+	private void removeContainerByElementId(String elementId) throws Exception {
+
+		Optional<Container> containerOptional = docker.getContainerByElementId(elementId);
+
+		if (containerOptional.isPresent()) {
+			String containerId = containerOptional.get().getId();
+			LoggingService.logInfo(MODULE_NAME, String.format("removing container \"%s\"", containerId));
+			try {
```
Comment (epankou) [inline daemon/src/org/eclipse/iofog/process_manager/ContainerManager.java]: move this part to separate method

File: daemon/src/org/eclipse/iofog/process_manager/ContainerManager.java [hunk 3]
```diff
@@ -163,41 +193,46 @@ private void updateContainer() throws Exception {
...
-				} catch (Exception e) {
-					LoggingService.logWarning(MODULE_NAME, e.getMessage());
-					return false;
+				if (elementOptional.isPresent()) {
+					try {
+						String containerId = addContainer(elementOptional.get());
+						result = new ContainerTaskResult(containerId, true);
+						break;
+					} catch (Exception e) {
+						result = new ContainerTaskResult(elementOptional.get().getContainerId(), false);
```
Comment (epankou) [inline daemon/src/org/eclipse/iofog/process_manager/ContainerManager.java]: take a look on containerId

File: daemon/src/org/eclipse/iofog/process_manager/ContainerTask.java [hunk 4]
```diff
@@ -28,34 +26,59 @@
         REMOVE
     }
 
-    public Tasks action;
-    public Object data;
-    public int retries;
+    private static final String EMPTY = "";
```
Comment (epankou) [inline daemon/src/org/eclipse/iofog/process_manager/ContainerTask.java]: library

File: daemon/src/org/eclipse/iofog/element/ElementManager.java [hunk 5]
```diff
@@ -132,16 +129,17 @@ public void clear() {
...
 
-	public Element getLatestElementById(String elementId) {
+	public Optional<Element> getLatestElementById(String elementId) {
 		return getLatestElementById(latestElements, elementId);
 	}
 
-	public Element getLatestElementById(List<Element> elements, String elementId) {
-		for (Element element : elements)
+	public Optional<Element> getLatestElementById(List<Element> elements, String elementId) {
+		Optional<Element> result = Optional.empty();
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/element/ElementManager.java]: use stream

**Comment by ilaryionava** [comment]
not relevant, outdated & repos moved

**Pull request closed without merging**